File: backend/app/models/scraping.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, Dict, Any
from uuid import UUID


class ScrapingJob(BaseModel):
    id: UUID
    status: str  # PENDING, RUNNING, COMPLETED, FAILED
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    funds_scraped: int = 0
    changes_detected: int = 0
    errors: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None

    class Config:
        from_attributes = True


class ScrapingTrigger(BaseModel):
    manual: bool = True
    funds_to_scrape: Optional[list[UUID]] = None  # None = all funds


class ScrapingStatus(BaseModel):
    is_running: bool
    current_job: Optional[ScrapingJob] = None
    last_run: Optional[datetime] = None
    next_scheduled_run: Optional[datetime] = None

File: frontend/src/types/scraping.ts
export interface ScrapingJob {
  id: string;
  status: 'PENDING' | 'RUNNING' | 'COMPLETED' | 'FAILED' | 'CANCELLED';
  started_at?: string;
  completed_at?: string;
  funds_scraped: number;
  changes_detected: number;
  errors?: string;
  metadata?: Record<string, unknown>;
}

export interface ScrapingStatus {
  is_running: boolean;
  current_job?: ScrapingJob;
  last_run?: string;
  next_scheduled_run?: string;
}

export interface ScrapingTrigger {
  manual: boolean;
  funds_to_scrape?: string[];
}
